src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -1136,7 +1136,9 @@ private async Task<Task> SetActiveTreeAnalysisTaskAsync(Func<Tuple<Task, Cancell
                         }
                     }
 
-                    await WaitForExecutingTaskAsync(executingTreeTask.Item1).ConfigureAwait(false);
+                    // Wait for the higher-priority operation to complete, and make sure to yield so its continuations
+                    // (which remove the operation from the collections) have a chance to execute.
```

Comment:
Unclear why we believe the other task is higher priority.

Reply:
It's higher priority because the continuations of `executingTreeTask` contain the code that removes items from the state collections, an operation which is necessary for the current operation to make progress. I was trying to write that, but open to suggestions on how to word this.

Reply:
Re-read the comment and it makes sense. Can't say why it didn't the first time :(

Reply:
I know the feeling